server/app/api.py

```diff
@@ -1191,24 +1192,100 @@ class SearchAPI(APIResource):
             }
         }
     }
+    
+    defaults = {}
+    
+    operators = {
+        'eq': op.__eq__,
+        'equal': op.__eq__,
+        'lt': op.__lt__,
+        'gt': op.__gt__,
+        'before': op.__lt__,
+        'after': op.__gt__
+    }
+    
+    flags = {
+        'user': lambda op, email:
+            UserAPI.model.query(
+                op(UserAPI.model.email, email)).get(),
+        'date': lambda op, s: datetime.datetime.strptime(s, '%Y-%m-%d'),
+        'onlyfinal': lambda op, boolean: boolean.lower() == 'true',
+        'assignment': lambda op, name:
+            AssignmentAPI.model.query(
+                op(AssignmentAPI.model.display_name, name)).get()
+    }
 
     def index(self, user, data):
-        blocks = SearchAPI.blockify(data['query'])
-        tokens = SearchAPI.tokenize(blocks)
-        print(tokens)
+        tokens = SearchAPI.tokenize(data['query'])
+        scope = SearchAPI.translate(tokens)
+        prime = SearchAPI.objectify(scope)
+        query = SearchAPI.querify(prime)
```

Comment:
I think this might be cleaner/easier to understand if the querify method querified the tokens and called objectify, which called translate, etc.